modules/gui/src/com/haulmont/cuba/gui/app/security/role/edit/tabs/UiPermissionsFrame.java

```diff
@@ -309,4 +311,16 @@ public void collapseTree() {
     public void expandTree() {
         componentsTree.expandTree();
     }
+
+    public void addComponentIdToComponentTextField() {
```

Comment:
I suppose that it would be useful to have an action that tracks the presence of the id attribute and disabled if no id is set.